### ISSUE TITLE
NAN and INF for training stage 1

### ISSUE THREAD

**Issue by yansuoyuli** [Yu-Qi-hang/ThinkRec#1]
Dear authors,

thanks for your amazing work!
I am wondering why the training stage 1 is nan or inf.

best regards

![Image](https://github.com/user-attachments/assets/9ebd8d4c-7216-4d5c-914c-987a0897a8fc)

**Comment by Yu-Qi-hang** [comment]
Which dataset did you use for the experiment? I had the same problem when I first used CoLLM on Book, but I solved it later. There are two possible causes for this phenomenon: the learning rate is too high, so you can try lowering it; or there is a problem with the data.

**Comment by yansuoyuli** [comment]
Yes, I change reason_mf_stage1.yaml train_ratios from [0.2, 0.8] to  [1.0], because there is only one dataset: ml-1m but two ratios

**Comment by yansuoyuli** [comment]
model:
  arch: mini_gpt4rec_v3
  model_type: pretrain_vicuna
  freeze_rec: True
  
  #stage1
  freeze_proj: True #    
  freeze_lora: False #
  prompt_path: "prompts/tallrec_amazon_.txt"

  # #stage2
  # freeze_proj: True
  # freeze_lora: True
  # prompt_path: "prompts/collm_amazon.txt"

  max_txt_len: 2048
  proj_token_num: 1
  proj_drop: 0
  proj_mid_times: 10
  end_sym: "###"
  prompt_template: '{}'
  llama_model: "models/Meta-Llama-3-8B-Instruct" #"/data/yuqihang/model/vicuna-7b/vicuna-7b-v0/"
  user_num: -100
  item_num: -100
  ans_type: 'v2'
  rec_model: "MF" #[MF, lightgcn,.....], see "Rec2Base" class in  minigpt4/models/rec_model.py
  loss_config:
    alpha: 0.1  #recommend
    beta: 0.9   #speaking
    theta: 0.9  #recommend
    gamma: 0.1   #speaking
  lora_config:
    use_lora: True
    r: 8
    alpha: 16
    target_modules: ["q_proj", "v_proj"] # ['lm_head'] ##["lm_head"] # ['lm_head'] ['lm_head'] #
    dropout: 0.05
  rec_config:
    user_num: -100
    item_num: -100
    embedding_size: 256
    pretrained_path: "ThinkRec/checkpoints/mf/0610_ml1m_best_model_d256_lr0.0001_wd1e-07.pth"
    

  
  # ckpt: /data/yuqihang/result/CoLLM/reproduce/20250405141/checkpoint_bestauc_uauc.pth

  

datasets:
  amazon_ood:
    path: ThinkRec/dataset/ml-1m/
    data_type: default
    build_info:
      seq_len: 10
      user2group: ''
      use_ids: False
      use_desc: True
      storage: ThinkRec/dataset/ml-1m

run:
  task: rec_pretrain
  # optimizer
  lr_sched: "linear_warmup_cosine_lr"  # "linear_warmup_step_lr"
  init_lr: 1e-4
  min_lr: 8e-5
  warmup_lr: 1e-5
  # init_lr: 3e-5
  # min_lr: 1e-5
  # warmup_lr: 1e-6
  mode: 'v3'

  weight_decay: 1e-3 #0.05
  max_epoch: 50
  iters_per_epoch: 800 #100 #50 #200
  batch_size_train: 1 #48
  batch_size_eval: 2 #48
  num_workers: 2
  warmup_steps: 200 #200

  seed: 42
  # output_dir: "output"
  output_dir: ThinkRec/reproduce

  amp: True
  resume_ckpt_path: null

  evaluate: False # False True
  train_splits: ["reason","train"]
  train_ratios: [1.0]
  valid_splits: ["valid"]
  test_splits: ["test"]
  # test_splits: ["test_warm", "test_cold", "test", "valid"]

  device: "cuda"
  world_size: 1
  dist_url: "env://"
  distributed: False

**Comment by yansuoyuli** [comment]
> Reason is the reasoning data synthesized from the sampled part of train. The ratio here controls the ratio of the two. If there is no reason and only train, you can modify it as follows:
> 
> # delete alpha,beta,gamma,theta
>   train_splits: ["train"]
>   valid_splits: ["valid"]
>   test_splits: ["test"]
> You can synthesize data by referring to `dataset/tools/getreflection.py` and process the results to `reason_ood2.pkl` by refering to the latest 2 section of `getreflection.ipynb` A fully setting is attached below:
> 
> {
>     "run": {
>         "task": "rec_pretrain",
>         "lr_sched": "linear_warmup_cosine_lr",
>         "init_lr": 0.0001,
>         "min_lr": 8e-05,
>         "warmup_lr": 1e-05,
>         "mode": "v3",
>         "weight_decay": 0.001,
>         "max_epoch": 1000,
>         "iters_per_epoch": 800,
>         "batch_size_train": 1,
>         "batch_size_eval": 2,
>         "num_workers": 2,
>         "warmup_steps": 200,
>         "seed": 42,
>         "output_dir": "/data/yuqihang/result/CoLLM/reproduce/ml1m/",
>         "amp": true,
>         "resume_ckpt_path": null,
>         "evaluate": false,
>         "train_splits": [
>             "train"
>         ],
>         "valid_splits": [
>             "valid"
>         ],
>         "test_splits": [
>             "test"
>         ],
>         "device": "cuda",
>         "world_size": 1,
>         "dist_url": "env://",
>         "distributed": false
>     },
>     "model": {
>         "arch": "mini_gpt4rec_v3",
>         "model_type": "pretrain_vicuna",
>         "freeze_rec": true,
>         "freeze_proj": true,
>         "freeze_lora": false,
>         "prompt_path": "prompts/tallrec_amazon.txt",
>         "max_txt_len": 2048,
>         "proj_token_num": 1,
>         "proj_drop": 0,
>         "proj_mid_times": 10,
>         "end_sym": "###",
>         "prompt_template": "{}",
>         "llama_model": "/data/yuqihang/model/Meta-Llama-3-8B-Instruct/",
>         "user_num": -100,
>         "item_num": -100,
>         "ans_type": "v2",
>         "rec_model": "MF",
>         "lora_config": {
>             "use_lora": true,
>             "r": 8,
>             "alpha": 16,
>             "target_modules": [
>                 "q_proj",
>                 "v_proj"
>             ],
>             "dropout": 0.05
>         },
>         "rec_config": {
>             "user_num": 5945,
>             "item_num": 3687,
>             "embedding_size": 256,
>             "pretrained_path": "/data/yuqihang/result/CoLLM/checkpoints/mf/0409_ml1m_best_model_d256_lr0.001_wd1e-06.pth"
>         },
>         "user2group": "/home/yuqihang/projects/CoLLM/collm-datasets/ml1mdu/user_group.csv"
>     },
>     "datasets": {
>         "amazon_ood": {
>             "path": "/home/yuqihang/projects/CoLLM/collm-datasets/ml1mdu/",
>             "data_type": "default",
>             "build_info": {
>                 "seq_len": 10,
>                 "cans_num": 1,
>                 "use_ids": false,
>                 "use_desc": true,
>                 "storage": "/home/yuqihang/projects/CoLLM/collm-datasets/ml1mdu/"
>             }
>         }
>     }
> }

Thanks for your reply. Can you share the processed data with reason data in google drive or baidu drive? Thanks

**Comment by Yu-Qi-hang** [comment]
Thank you very much for your interest. We plan to gradually release the processed data and trained weights within three months.

**Comment by yansuoyuli** [comment]
how about the /home/yuqihang/projects/CoLLM/collm-datasets/bookdu/reflection/wrong_1.txt? how can I get this file?
